packages/app-mobile/components/NotesBar.tsx

```diff
@@ -0,0 +1,278 @@
+import * as React from 'react';
+import { View, Text, StyleSheet, TextInput, TouchableOpacity, FlatList } from 'react-native';
+import { State } from '@joplin/lib/reducer';
+import { themeStyle } from './global-style';
+const { connect } = require('react-redux');
+const Icon = require('react-native-vector-icons/Ionicons').default;
+const { _ } = require('@joplin/lib/locale');
+import { Style } from './global-style';
+import Note from '@joplin/lib/models/Note';
+import NotesBarListItem from './NotesBarListItem';
+import Folder from '@joplin/lib/models/Folder';
+
+interface Props {
+    themeId: string;
```

Comment:
```suggestion
    themeId: number;
```
It looks like [`themeStyle` takes a number!](https://github.com/laurent22/joplin/blob/dev/packages/lib/theme.ts#L369)

Reply:
It's a bit tricky, but `themeStyle` is also defined [here](https://github.com/laurent22/joplin/blob/a5e6491cda3548537d5d032aa63e7e8bc34eceac/packages/app-mobile/components/global-style.js#L91) in `global-style.js`, there's another `themeStyle` function [ defined in theme.ts](https://github.com/laurent22/joplin/blob/a5e6491cda3548537d5d032aa63e7e8bc34eceac/packages/lib/theme.ts#L369) (The one you referred to), but that's not the function used here. The `themeStyle` defined in global-style.js ultimately plugs in the `theme` paramater into `themeById` from `theme.ts`, which takes in a string.

I refactored global-style.js into typescript in this pr, I made the mistake of making themeStyle's parameter a number at first, but corrected it after. I hope that all makes sense  😅

Reply:
It's a number, or it should be one. How is it in other places in the codebase?

Reply:
It should be a string, because the parameter is being passed to [themeById function](https://github.com/laurent22/joplin/blob/a5e6491cda3548537d5d032aa63e7e8bc34eceac/packages/lib/theme.ts#L26) (whose parameter is of type string). It's being passed into themeById [here](https://github.com/laurent22/joplin/blob/a5e6491cda3548537d5d032aa63e7e8bc34eceac/packages/app-mobile/components/global-style.js#L100).

Reply:
That's a mistake in theme.ts then. You can see that eg "Setting.THEME_LIGHT" is a number so theme ID should be a number.

Reply:
Oh. alright